cpp/oneapi/dal/algo/knn/detail/model_impl.hpp

```diff
@@ -33,4 +34,4 @@ class detail::model_impl : public base {
     interop_model* interop_;
```

Comment:
There is no delete for interop model. It looks like memory leak. That's what I've found in KD-tree method training:
```
auto interop = new daal_interop_model_t(model_ptr);
const auto model_impl = std::make_shared<detail::model_impl>(interop);
```